packages/republik/graphql/resolvers/_mutations/resubscribeEmail.js

```diff
@@ -0,0 +1,84 @@
+const { Roles } = require('@orbiting/backend-modules-auth')
+const MailchimpInterface = require('../../../../mail/MailchimpInterface')
+const logger = console
+
+const {
+  cache: { create },
+} = require('@orbiting/backend-modules-utils')
+
+const QUERY_CACHE_TTL_SECONDS = 60 // One minute
+
+module.exports = async (_, args, context) => {
+  const { userId } = args
+  const {
+    user: me,
+    pgdb,
+    req,
+    t,
+    mail: { getNewsletterSettings },
+  } = context
+
+  const user = userId ? await pgdb.public.users.findOne({ id: userId }) : me
+
+  if (!user) {
+    console.error('user not found', { req: req._log() })
+    throw new Error(t('api/users/404'))
+  }
+
+  const { id, email, firstName, lastName } = user
+
+  Roles.ensureUserIsMeOrInRoles(user, me, ['supporter'])
+
+  const mailchimp = MailchimpInterface({ logger })
+  const member = await mailchimp.getMember(email)
+
+  const body = {
+    email_address: email,
+  }
+
+  if (!member) {
+    body.status_if_new = MailchimpInterface.MemberStatus.Subscribed
+    body.status = MailchimpInterface.MemberStatus.Subscribed
+    body.merge_fields = {
+      FNAME: firstName || '',
+      LNAME: lastName || '',
+    }
+    await mailchimp.updateMember(email, body)
```

Comment:
It there is no `member`, it should `updateMember` (hence create member) and then `return getNewsletterSettings({ user })` – no need to walk over cache locking and subsequent `if` stmts.

Reply:
I've put cache locking and subsequent if stmts in else case and return `getNewsletterSettings({ user })` at the end